packages/flutter/lib/src/material/data_table.dart

```diff
@@ -851,15 +876,34 @@ class DataTable extends StatelessWidget {
       ?? dataTableTheme.dataRowHeight
       ?? themeData.dataTableTheme.dataRowHeight
       ?? kMinInteractiveDimension;
+
+    String semanticsLabel = '';
+    if (rowHeading != null) {
+      semanticsLabel += '$rowHeading,';
+    }
+    if (columnHeading != null) {
+      semanticsLabel += '$columnHeading,';
+    }
     label = Container(
       padding: padding,
       height: effectiveDataRowHeight,
       alignment: numeric ? Alignment.centerRight : AlignmentDirectional.centerStart,
-      child: DefaultTextStyle(
-        style: effectiveDataTextStyle.copyWith(
-          color: placeholder ? effectiveDataTextStyle.color!.withOpacity(0.6) : null,
-        ),
-        child: DropdownButtonHideUnderline(child: label),
+      child: Stack(
```

Comment:
why do you need a stack? it looks like the second child is for semantics only?

Reply:
Because if I put 'Row i of # totalRows, Column j of # totalColumns" as a label in the original Semantics widget which is the parent of the DefaultTextStyle it will be announced before the information in the DataCell thus had to add another Semantics widget to make sure its announced after the DataCell info itself. Please feel free to suggest better alternatives. Thanks!